components/ui/TextInput.tsx

```diff
@@ -28,40 +31,44 @@ export const TextInput: FC<TextInputProps> = ({
   };
 
   return (
-    <div className={"flex flex-grow flex-col"}>
+    <div className="mb-4 flex flex-grow flex-col">
       <label
         htmlFor={inputName}
         className={tw(
           "mb-2 block text-sm text-gray-800 dark:text-white",
           labelClassName
         )}
       >
-        {upperFirst(inputName)}
+        {upperFirst(words(inputName).join(" "))}
```

Comment:
I think you're complicating things unnecessarily here. Just pass in id (or input name if you prefer) and label separately instead of trying to force one prop into several ones.